packages/web-app-pdf-viewer/src/App.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <main>
+    <loading-screen v-if="loading" />
+    <error-screen v-else-if="loadingError" :message="errorMessage" />
+    <object
+      v-else
+      class="pdf-viewer oc-width-1-1 oc-height-1-1"
+      :data="blobUrl"
+      type="application/pdf"
+    />
+  </main>
+</template>
+<script>
+import { mapGetters } from 'vuex'
+import { useAppDefaults } from 'web-pkg/src/composables'
+import { DavProperties } from 'web-pkg/src/constants'
+import { buildResource } from '../../web-app-files/src/helpers/resources'
+import HTTPError from 'owncloud-sdk'
+import ErrorScreen from './components/ErrorScreen.vue'
+import LoadingScreen from './components/LoadingScreen.vue'
+
+export default {
+  name: 'PDFViewer',
+  components: {
+    ErrorScreen,
+    LoadingScreen
+  },
+  setup() {
+    return {
+      ...useAppDefaults({
+        applicationName: 'pdf-viewer'
+      })
+    }
+  },
+  data: () => ({
+    loading: true,
+    loadingError: false,
+    errorMessage: '',
+    filePath: '',
+    blobUrl: ''
+  }),
+  computed: {
+    ...mapGetters(['getToken'])
+  },
+  async created() {
+    this.filePath = this.currentFileContext.path
+
+    try {
+      let resource = await this.getFileInfo(
```

Comment:
Can be simplified by using `getFileContents` which you made available via `useAppDefaults`, right?

Reply:
The problem is that `getFileContents` returns text instead of blob.

Reply:
Then let's add handling of `blob` as `resolveType` here

https://github.com/owncloud/owncloud-sdk/blob/b5df7ff922bfc9c1d70763267335df982331cbd4/src/helperFunctions.js#L343

This kind of logic should never be handled in an app imho

Reply:
[1028](https://github.com/owncloud/owncloud-sdk/pull/1028)

Reply:
@diocas Would you like me to take this over and adjust to that? (also some routing changes in my pipeline once again)

Reply:
@pascalwengerter feel free to take it, thanks (just add the cern tag to the new PR please)

Reply:
Thanks for the feedback @diocas, it'll be @dschmidt who'll take over from here I think 🐎